### DISCUSSION TITLE
Discussions in issue vs PR

### DISCUSSION THREAD

**Discussion by stephan-herrmann** [eclipse-jdt/eclipse.jdt.core#2630] [General]
We don't seem to have a clear strategy how to use the dichotomy of issues vs pull request.

This came up in the [discussion of where to set milestones](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2413#issuecomment-2184033231)  (which led to #2627).

But then it also concerns the risk of [one discussion being split between two locations](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2413#issuecomment-2184931482).

**Comment by stephan-herrmann** [comment]
@srikanth-sankaran  wrote:
> The dichotomy between PR and Issue also very often ends up bifurcation the discussions which is unfortunate.

good point (and I feel being looked at :) ).

We could try to be more strict in separating:
* issue to discuss the problem
* PR to discuss the solution?

But I'm not sure the the boundary can always be drawn cleanly.

Another option:
* issue is only for communication with users
* all discussion among the team goes into PRs

This would push us towards creating a PR with at least a failing test as soon as possible, so we *have* a PR to use for that discussion.

Having all discussion in the issue doesn't work if we want to use code-centric reviews (which may concern more than matters of code style, might trigger general discussions of strategy).

Other ideas?

**Comment by stephan-herrmann** [comment]
Mildly related: As a huge fan of hypertext and cross-linking I can't stop being baffled by email notifications sent by github containing links to 404, and likewise references to commits that once existed and no longer do.

**Comment by stephan-herrmann** [comment]
Pragmatic approach:

Once a PR is created for an issue, we could add a clear mark to the issue, that all discussion (should!) continue in the PR.

**Comment by jukzi** [comment]
my 50 cent: Use the PR for code review only, keep discussing general questions on the topic in the Issue. Most times question about the issue should have been completed before a PR, but there may be several PRs per issue

> email notifications sent by github

Please complain about that to the github support https://docs.github.com/en/support/contacting-github-support
